spec/controllers/admins/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Admins::OmniauthCallbacksController, type: :controller do
+  describe "allows logged in admins" do
+    context 'if admin' do
+      let!(:google_admin) { create(:admin, provider: 'google',
+        uid: "1076060026507") }
+      let(:auth) { OmniAuth::AuthHash.new(
+        provider: google_admin.provider,
+        uid: google_admin.uid,
+        info: {
+          email: google_admin.email,
+          login: google_admin.login
+        }
+      )}
+
+      it 'return Admin with google' do
+        expect(Admin.from_omniauth(auth)).to be_persisted
```

Comment:
extract Admin.from_omniauth(auth) into variables
because it's 3 times in the code

Reply:
ok